airbyte-cdk/python/airbyte_cdk/sources/streams/http/http.py

```diff
@@ -190,9 +203,8 @@ def _send_request(self, request: requests.PreparedRequest) -> requests.Response:
         Unexpected transient exceptions use the default backoff parameters.
         Unexpected persistent exceptions are not handled and will cause the sync to fail.
         """
-        response: requests.Response = self._session.send(request)
+        response: requests.Response = self._session.send(request, **request_kwargs)
```

Comment:
I see these changes as potentially destructive.
The problem is that we now provide multiple ways for users to define request parameters (headers, path, params, auth). If user will do this then he needs to reimplement the logic of request_params, request_headers, etc, as there is no way to know default values in request_kwargs call.

Reply:
ok, I checked the source code and it seems that the only parameters that we can use here are:
* allow_redirects
* stream
* timeout
* verify
* cert
* proxies
So there is no conflict with request args, but it still not possible to customize `auth` (FYI)